src/routes/download/drs.service.ts

```diff
@@ -0,0 +1,70 @@
+import urlJoin from 'url-join';
+import fetch from 'node-fetch';
+
+import { AppConfig } from '@/config/global';
+import logger from '@/logger';
+
+// types from https://ga4gh.github.io/data-repository-service-schemas/preview/release/drs-1.0.0/docs/#_accessmethod
+// will only be using https
+enum AccessType {
+  HTTPS = 'https',
+  S3 = 's3',
+  GS = 'gs',
+  FTP = 'ftp',
+  FILE = 'file',
+  GSIFTP = 'gsiftp',
+  GLOBUS = 'globus',
+  HTSGET = 'htsget',
+}
+
+interface AccessMethod {
+  type: AccessType;
+  access_url: { url: string };
+  region: string;
+}
+
+export interface DRSMetada {
+  name: string;
+  accessUrls: AccessMethod[];
+}
+
+export const parseDRSPath = (filepath: string, config: AppConfig) => {
+  const toUri = new URL(filepath);
```

Comment:
Not sure if this is cleaner but just had the thought: `const {hostname, pathname: objectId} = new URL(filepath);`

Reply:
More importantly:

There is no way to know that the pathname is the objectId. it's just the path of whatever filepath is provide. If you need the object ID make that a parameter on its own.

Furthering this line of thought, there are many `filepath` values (of type string) that will break this method and throw errors.

I would propose a method signature that asks for the pieces you need: `buildDRSUrl(domain: string, objectId: string) { // drs objectPath and protocol are pulled from config directly. }`

A second method that actually parses the filepath would look like:
`parseDRSPath(filepath: string): {domain: string, objectId: string} => { // attempt to do new URL with try/catch }`